From 44e37774f1bf639d5c65744282f58d09cdbbbd55 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Tue, 26 Nov 2019 15:41:08 +0100
Subject: [PATCH] Fix bootcount issues due to missing commands

fatwrite and 'env export' commands did not work
because they were being undefined or were not defined
at all. Setting them in the board defconfig appears
to be not enough, some need to be present in the
board config file.

Upstream-status: Inappropriate[configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 configs/mx8mq_phanbell_defconfig  | 2 ++
 include/configs/imx8mq_phanbell.h | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/configs/mx8mq_phanbell_defconfig b/configs/mx8mq_phanbell_defconfig
index 792cfe5b2a..c55b8c2b77 100644
--- a/configs/mx8mq_phanbell_defconfig
+++ b/configs/mx8mq_phanbell_defconfig
@@ -43,3 +43,5 @@ CONFIG_ZIRCON_BOOT_IMAGE=y
 CONFIG_SD_BOOT=y
 CONFIG_CMD_IMPORTENV=y
 CONFIG_CMD_PART=y
+CONFIG_CMD_EXPORTENV=y
+CONFIG_FAT_WRITE=y
diff --git a/include/configs/imx8mq_phanbell.h b/include/configs/imx8mq_phanbell.h
index 5170c54078..e03a4793a9 100644
--- a/include/configs/imx8mq_phanbell.h
+++ b/include/configs/imx8mq_phanbell.h
@@ -70,7 +70,7 @@
 /* Flat Device Tree Definitions */
 #define CONFIG_OF_BOARD_SETUP
 
-#undef CONFIG_CMD_EXPORTENV
+
 #undef CONFIG_CMD_IMLS
 
 #undef CONFIG_CMD_CRC32
@@ -233,6 +233,7 @@
 #define CONFIG_CMD_EXT4
 #define CONFIG_CMD_EXT4_WRITE
 #define CONFIG_CMD_FAT
+#define CONFIG_FAT_WRITE
 
 #define CONFIG_SUPPORT_EMMC_BOOT	/* eMMC specific */
 #define CONFIG_SYS_MMC_IMG_LOAD_PART	1
-- 
2.17.1

